README.md

```diff
@@ -277,28 +278,160 @@ This gives the start timestamps that can be used to correlate the events and the
 
 ## Unit tests
 
-Unit tests can be run with `go test` outside of Docker if you have the correct C++ dependencies installed on your system.
+ 0chain unit tests verify the behaviour of individual parts of the program. A config for the base docker image can be provided on run to execute general unit tests.
+
+
+![unit testing uml](https://user-images.githubusercontent.com/65766301/120052862-0b4ffd00-c045-11eb-83c8-977dfdb3038e.png)
+
+### Getting started
+
+### Prerequisites
+
+Docker and Git must be installed in order to run the tests .
+
+Install Git using the following command:
 
 ```
-$ cd code/go/0chain.net/my-pkg
-$ go test
+sudo apt install git
 ```
 
-Otherwise, we have a Docker image which takes care of installing the build dependencies for you in an environment identical to our other Docker builds.
+Docker installation instructions can be found [here](https://docs.docker.com/engine/install/).
+
+### Cloning the repository and Building Base Image
 
-First build the base image.
+Clone the 0chain repository:
 
 ```
-$ ./docker.local/bin/build.base.sh
+git clone https://github.com/0chain/0chain.git
+```
+
+Navigate to 0chain folder and run the script to build base docker image for unit testing :
+
+```
+cd 0chain
+./docker.local/bin/build.base.sh
+```
+
+The base image includes all the dependencies required to test the 0chain code.
+
+### Running Tests
+
+Now run the script containing unit tests .
+
+```
+./docker.local/bin/unit_test.sh 
+```
+
+The list of packages is optional, and if provided runs only the tests from those packages. Command for running unit tests with specific packages .
+
+```
+./docker.local/bin/unit_test.sh [<packages>]
 ```
 
-Then run the tests.
+###  Testing Steps
+
+Unit testing happens over a series of steps one after the other.
+
+#### Step 1: FROM zchain_build_base
```

Comment:
I am not sure you want to go into this much detail. Readers of the main readme are just going to want to know how to run unit tests, not the details of how the script works. This would be better in `0chian/docker.local/buld.unit_test`.